PeakCoreData/Core/Operations/CoreDataBatchDeleteOperation.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  CoreDataBatchDeleteOperation.swift
+//  PeakCoreData
+//
+//  Created by David Yates on 21/05/2020.
+//  Copyright © 2020 3Squared Ltd. All rights reserved.
+//
+
+import CoreData
+
+/// Performs an `NSBatchDeleteRequest` and returns an array of deleted `NSManagedObjectID`s.
+open class CoreDataBatchDeleteOperation<Entity: ManagedObjectType>: CoreDataOperation<[NSManagedObjectID]> {
+    
+    private let predicate: NSPredicate?
+    
+    public init(predicate: NSPredicate? = nil, persistentContainer: NSPersistentContainer, mergePolicyType: NSMergePolicyType = .mergeByPropertyObjectTrumpMergePolicyType) {
+        self.predicate = predicate
+        super.init(persistentContainer: persistentContainer, mergePolicyType: mergePolicyType)
+    }
+    
+    open override func performWork(in context: NSManagedObjectContext) {
+        let request = NSFetchRequest<NSFetchRequestResult>(entityName: Entity.entityName)
+        request.predicate = predicate
+        let deleteRequest = NSBatchDeleteRequest(fetchRequest: request)
+        deleteRequest.resultType = .resultTypeObjectIDs
+        do {
+            let result = try context.execute(deleteRequest) as! NSBatchDeleteResult
+            let objectIDArray = result.result as! [NSManagedObjectID]
+            let changes = [NSDeletedObjectsKey: objectIDArray]
+            NSManagedObjectContext.mergeChanges(fromRemoteContextSave: changes, into: [context])
```

Comment:
If `context` is a background context, do you need to `mergeChanges`? It'll be disposed of pretty shortly. Or does doing this cause them to get pushed back up to the `viewContext`?